src/decorators.ts

```diff
@@ -451,6 +459,26 @@ export function getOrCreateClassRegistration<CT extends { new (...args: unknown[
   return clsReg;
 }
 
+export function associateClassWithEventReceiver<CT extends { new (...args: unknown[]) : object }>(
```

Comment:
Isn't this really `registerEventReceiver`?  If not, why not?

Reply:
It is not quite `registerEventReceiverClass`.  Yes, it does register the event receiver class (if it isn't registered already); that's not an explicit step because we wanted magic.  (Where "event receiver class" is the class that provides the listeners or pollers or whatnot.)

This call also associates the event receiver with an application class ... think of `@Kafka` or `@KafkaConsume` which don't really create kafka in the first place, but associate it with application classes and functions.

So the full name (which I thought long enough already) could be `registerApplicationClassWithEventReceiverClass` or `registerClassWithEventReceiver` but I picked `associateClassWithEventReceiver` because you probably also need a registered method for anything to really happen.

Reply:
Makes sense!